src/main/java/com/joyent/manta/cosbench/MantaStorage.java

```diff
@@ -241,9 +265,102 @@ public void createObject(final String container,
         }
     }
 
+    /**
+     * Method for parsing and uploading to use multipart upload.
+     * @param data - the input stream
+     * @param path - the path of
+     * @throws IOException - this has the possiblity of throwing a IOException
+     */
+    private void multipartUpload(final InputStream data, final String path) throws IOException {
+        logger.debug("Client is not encrypted!");
+        ServerSideMultipartManager multipartManager = new ServerSideMultipartManager(client);
+        if (splitSize == null) {
+            splitSize = new Integer(DEFAULT_SPLIT);
+        }
+        logger.info("Object Size : " + this.objectSize);
+        logger.info("Split Size : " + splitSize);
+        int splits = objectSize / splitSize;
+        int remainder = objectSize % splitSize;
+        if (remainder > 0) {
+            splits++;
+        }
+        MantaMultipartUploadTuple[] parts = new MantaMultipartUploadTuple[splits];
+        // ServerSideMultipartUpload upload =
+        // multipartManager.initiateUpload(path);
+        ServerSideMultipartUpload upload = multipartManager.initiateUpload(path);
+
+        logger.debug("Multipart upload initiated");
+        byte[] content = IOUtils.toByteArray(data);
```

Comment:
This is rather dangerous to do for large object sizes. You are loading the entire object into memory here. For example, if you had 20mb objects and you wanted to simulate 1,000 concurrent uploads, this method alone would consume around 20gb of memory. Even if you have that much memory, preemptively reading the whole stream into memory before uploading will cause a delay and mess up your measurement.

One important thing to note is the COSBench measures timings directly from the reads from the data `InputStream`.

You will probably want to rewrite the logic in your for loop to do something like wrap the data `InputStream` in a `BoundedInputStream` set to the split size and then upload that `BoundedInputStream` as a single part and then iterate that way until you have uploaded the entirety of the source data `InputStream` as a multipart upload.